addons/web/static/src/views/kanban/kanban_model.js

```diff
@@ -562,6 +562,39 @@ export class KanbanDynamicGroupList extends DynamicGroupList {
         ]);
         await Promise.all(this.groups.map((group) => group.updateProgressData(progressData)));
     }
+
+    getAdditionalFields() {
+        return {};
+    }
+
+    async createGroupWithFoldDetail(model, fields) {
+        const id = await this.model.orm.create(model, [fields]);
+        return super.getGroup(id, fields['name'], fields['fold']);
+    }
+
+    async createGroup() {
+        const fold = arguments[1];
+        if (fold != null) {
+            const model = this.model.root.fields[this.model.root.groupBy].relation;
+            let fields = {
+                'name': arguments[0],
+                'fold': arguments[1],
+            }
+            const additional_fields = this.getAdditionalFields(...arguments);
+            if (additional_fields === false) {
+                return await super.createGroup(...arguments);
+            }
+            else {
+                if (additional_fields) {
+                    fields = {...fields, ...additional_fields};
+                }
+                return await this.model.mutex.exec(() => this.createGroupWithFoldDetail(model, fields));
+            }
+        }
+        else {
+            return await super.createGroup(...arguments);
+        }
+    }
```

Comment:
```suggestion
    async createGroup() {
        const [ name, fold ] = arguments;
        if (fold != undefined) {
            return await this.model.mutex.exec(() => this.createGroupWithFoldDetail(
                this.model.root.fields[this.model.root.groupBy].relation,
                { name, fold, ...this.getAdditionalFields() },
            ));
        }
        return await super.createGroup(...arguments);
    }
```
Doesn't that work (if `getAdditionalFields` returns an empty dict i/o false)? :thinking:

Reply:
why did you suggest changing fold != null -> fold != undefined ?